data/src/main/java/ch/srg/dataProvider/integrationlayer/data/ImageUrl.kt

```diff
@@ -19,15 +19,25 @@ data class ImageUrl(
      *
      * @return the undecorated url
      */
-    val rawUrl: String
+    internal val rawUrl: String
 ) : Serializable {
 
-    @JvmOverloads
-    fun getIlImage(): IlImage {
-        return IlImage(rawUrl)
+    /**
+     * Url
+     *
+     * @param decorator The [ImageUrlDecorator] used to decorate the [rawUrl].
+     * @param widthPixels The width of the image.
+     * @return The decorated [rawUrl].
+     */
+    fun url(decorator: ImageUrlDecorator, widthPixels: Int): String {
+        return decorator.decorate(rawUrl, widthPixels)
     }
 
     override fun toString(): String {
         return rawUrl
     }
 }
+
+interface ImageUrlDecorator {
+    fun decorate(source: String, widthPixels: Int): String
```

Comment:
If you make this a `fun interface`, and invert the order of the arguments of `ImageUrl.url()`, you can call it as follow:
```kotlin
val url = ImageUrl("https://rts.ch/")
url.url(widthPixels = 1920) { source, widthPixels ->
    "https://rts.ch/new-url"
}
```

```suggestion
fun interface ImageUrlDecorator {
    fun decorate(source: String, @Px widthPixels: Int): String
```